### PR TITLE
feat: Improve models

### PR DESCRIPTION
**Pull request by nicola-smartive**

### CODE REVIEW

File: src/models.ts [hunk 1]
```diff
@@ -28,19 +22,11 @@ export type RawEnumModel = BaseModel & { type: 'raw-enum'; values: string[] };
 export type InterfaceModel = BaseModel & { type: 'interface'; fields: ModelField[] };
 
 export type RawObjectModel = BaseModel & {
-  type: 'raw-object';
+  type: 'raw';
   fields: ModelField[];
-  rawFilters?: { name: string; type: string; list?: boolean; nonNull?: boolean }[];
 };
 
-export type JsonObjectModel = BaseModel & {
```
Comment (nicola-smartive) [inline src/models.ts:36]: new: just use 'raw'

File: src/models.ts [hunk 2]
```diff
@@ -55,13 +41,11 @@ export type MutationHook = (
 export type ObjectModel = BaseModel & {
   type: 'object';
   interfaces?: string[];
-  // createdAt, createdBy, updatedAt, updatedBy can be null
-  nonStrict?: boolean;
```
Comment (nicola-smartive) [inline src/models.ts:59]: can be achieved using optional creatable/updatable/deletable field overrides below

File: src/models.ts [hunk 3]
```diff
@@ -55,13 +41,11 @@ export type MutationHook = (
...
   queriable?: boolean;
   listQueriable?: boolean;
-  creatable?: boolean;
-  updatable?: boolean;
-  deletable?: boolean;
+  creatable?: boolean | { createdBy?: Partial<ModelField>; createdAt?: Partial<ModelField> };
```
Comment (thommymo) [inline src/models.ts:46]: Not sure about the naming: 
```suggestion
  creatable?: boolean | { byRole?: Partial<ModelField>; createdAt?: Partial<ModelField> };
```

File: src/models.ts [hunk 4]
```diff
@@ -71,125 +55,92 @@ export type ObjectModel = BaseModel & {
...
-  type: string;
-  default?: Value;
   list?: boolean;
   nonNull?: boolean;
+  default?: Value;
   args?: Field[];
   directives?: Directive[];
-};
+} & (
+  | ({ fieldType?: 'primitive' } & (
```
Comment (thommymo) [inline src/models.ts:72]: Nice change!

File: src/models.ts (hunk 3 above)
Comment (thommymo) [reply src/models.ts:46]: Ah, one moment, this is not the role, this is the actual user id, right?

File: src/models.ts [hunk 5]
```diff
@@ -71,125 +55,92 @@ export type ObjectModel = BaseModel & {
...
+          dateTimeType?: 'year' | 'date' | 'datetime' | 'year_and_month';
+          endOfDay?: boolean;
+        }
+      | ({
+          type: 'Int';
+          intType?: 'currency';
+        } & BaseNumberType)
+      | ({
+          type: 'Float';
+          foatType?: 'currency' | 'percentage';
```
Comment (dwirz) [inline src/models.ts:92]:
```suggestion
          floatType?: 'currency' | 'percentage';
```

File: src/models.ts [hunk 6]
```diff
@@ -28,19 +22,11 @@ export type RawEnumModel = BaseModel & { type: 'raw-enum'; values: string[] };
 export type InterfaceModel = BaseModel & { type: 'interface'; fields: ModelField[] };
 
 export type RawObjectModel = BaseModel & {
-  type: 'raw-object';
+  type: 'raw';
   fields: ModelField[];
-  rawFilters?: { name: string; type: string; list?: boolean; nonNull?: boolean }[];
```
Comment (dwirz) [inline src/models.ts:33]: How does this get handled now?

**Pull request merged**